utils/report.c

```diff
@@ -796,8 +936,24 @@ void (*setup_default_field_func[])(struct list_head*, struct opts*,
 void setup_report_field(struct list_head *output_fields, struct opts *opts,
 			enum avg_mode avg_mode)
 {
+	struct display_field **f_table;
+	if (opts->diff) {
+		if (opts->diff_policy && diff_policy.full) {
+			if (diff_policy.percent)
+				f_table = field_diff_full_percent_table;
+			else
+				f_table = field_diff_full_table;
+		}
+		else {
+			f_table = field_diff_table;
+		}
+	}
+	else {
+		f_table = field_table;
+	}
+
 	setup_field(output_fields, opts, setup_default_field_func[avg_mode],
-		    field_table, ARRAY_SIZE(field_table));
+		    f_table, ARRAY_SIZE(field_table));
```

Comment:
Shouldn't `ARRAY_SIZE()` be changed as well?

Reply:
you have hawkeye. I tried to avoid to use length variable for it. but it's modified. thank you.